File: 003/DigitCounts.java
class Solution {
    /*
     * param k : As description.
     * param n : As description.
     * return: An integer denote the count of digit k in 1..n
     */
    public int digitCounts(int k, int n) {
        // write your code here
        int count = 0;
        for (int i=0; i<=n; i++) {
            String str = i+"";
            for(int j=0;j<str.length(); j++) {
                if(str.charAt(j) == 48+k) {
                    count ++;
                }
            }
        }
        return count;
    }
};

File: 003 统计数字/DigitCounts.cpp
#include<iostream>
using namespace std;

class Solution {
public:
	int digitCounts(int k, int n) {
		// write your code here
		int count = 0;
		for (int i = 0; i <= n; i++)
		{
			count += countTheNumber(k, i);
		}
		return count;
	}
	int countTheNumber(int k, int num) {
		int count = 0;
		while (num >= 10)
		{
			if (num % 10 == k) count++;

			num = num / 10;
		}
		if (num == k) count++;
		return count;
	}
};

int main() {
	Solution s;
	cout << s.digitCounts(0, 19) << "\n";
}

